log_forwarded_for_ip/__init__.py

```diff
@@ -3,13 +3,24 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 from werkzeug.serving import WSGIRequestHandler
+from mimetools import Message
+from openerp import models
+from .hooks import restore_address_string
 
 
 def address_string(self):
-    forwarded_for = self.headers.get('X-Forwarded-For', '').split(',')
-    if forwarded_for and forwarded_for[0]:
-        return forwarded_for[0]
-    else:
-        return self.client_address[0]
+    if self.headers and isinstance(self.headers, Message):
+        forwarded_for = self.headers.get('X-Forwarded-For', '').split(',')
+        if forwarded_for and forwarded_for[0]:
+            return forwarded_for[0]
+    return self.client_address[0]
 
-WSGIRequestHandler.address_string = address_string
+
+class log_forwarded_for_ip_installed(models.AbstractModel):
```

Comment:
even though it's tiny, please do that in `models/log_forwarded_for_ip_installed.py`